src/clinical/service-worker-thread/tasks.ts

```diff
@@ -269,22 +276,20 @@ function extractEntityDataFromDonors(
     });
   });
 
-  const completionStats: CompletionRecord[] = donors
-    .map(({ completionStats, donorId, specimens }): CompletionRecord | undefined => {
-      let completionRecord =
-        completionStats && donorId ? { ...completionStats, donorId } : undefined;
+  const completionStats: CompletionDisplayRecord[] = donors
+    .map(({ completionStats, donorId, specimens: donorSpecimenRecords }):
+      | CompletionDisplayRecord
+      | undefined => {
+      const filteredSpecimenData = donorSpecimenRecords.filter(dnaSampleFilter);
+      const specimens = { specimens: calculateSpecimenCompletionStats(filteredSpecimenData) };
+
+      const completionRecord: CompletionDisplayRecord | undefined =
+        completionStats && donorId
+          ? { ...completionStats, donorId, entityData: specimens }
+          : undefined;
 
       // Update Completion Stats to display Normal/Tumour stats
       if (completionRecord && completionRecord.coreCompletion?.specimens > 0) {
-        const donorSpecimenData = specimens.filter(dnaSampleFilter);
-
-        completionRecord = {
-          ...completionRecord,
-          coreCompletion: {
-            ...completionRecord.coreCompletion,
-            ...calculateSpecimenCompletionStats(donorSpecimenData),
-          },
-        };
       }
```

Comment:
is this an empty if block?!

Reply:
Yes, didn't remove that properly I guess. This is now removed, with an updated comment. We can put the if block back if we don't want to populate every response